### PR TITLE
Tobit

### PR DESCRIPTION
**Pull request by kels271828**
Adding tobit model for data generated by a censored Gaussian distribution.

### CODE REVIEW

**Review by zhengp0 — COMMENTED** [review]
Thanks @kels271828! Let's discuss this during tomorrow's meeting.

File: src/regmod/models/tobit.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+        Returns
+        -------
+        float
+            Negative log likelihood.
+
+        """
+        weights = self.data.weights*self.data.trim_weights
+        obj_param = weights*self.get_nll_terms(coefs)
+        return obj_param.sum() + self.objective_from_gprior(coefs)
```
Comment (zhengp0) [inline src/regmod/models/tobit.py:107]: `get_nll_terms` have the data weights, here we only need to apply the trim weights

File: src/regmod/models/tobit.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+        """
+        vals = {
+            "mu": params[0],
+            "sigma": params[1],
+            "y": self.data.obs,
+            "nll_terms": jnp.zeros(self.data.num_obs)
+        }
+        vals = lax.fori_loop(0, self.data.num_obs, _nll_term, vals)
+        return vals["nll_terms"]
```
Comment (zhengp0) [inline src/regmod/models/tobit.py]: Can we vectorize this?

Comment (zhengp0) [reply src/regmod/models/tobit.py]: Maybe the `vmap` as well

File: src/regmod/models/tobit.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        params : list[array_like]
+            Model parameters.
+
+        Returns
+        -------
+        list[DeviceArray]
+            Derivatives of negative log likelihood.
+
+        """
+        return grad(lambda pars: jnp.sum(self.nll(pars)))(params)
```
Comment (zhengp0) [inline src/regmod/models/tobit.py:180]: Can we directly redefine `jacobian2`?

File: src/regmod/models/tobit.py (hunk 1 above)
Comment (kels271828) [reply src/regmod/models/tobit.py:107]: Fixed!

File: src/regmod/models/tobit.py (hunk 2 above)
Comment (kels271828) [reply src/regmod/models/tobit.py]: Updated!

File: src/regmod/models/tobit.py (hunk 3 above)
Comment (kels271828) [reply src/regmod/models/tobit.py:180]: Probably, we can put this on the wishlist

File: src/regmod/models/tobit.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+        DeviceArray
+            Terms of negative log likelihood.
+
+        """
+        y = self.data.obs
+        mu = params[0]
+        sigma = params[1]
+        pos_term = jnp.log(sigma) - logpdf((y - mu)/sigma)
+        npos_term = -logcdf(-mu/sigma)
+        return jnp.where(y > 0, pos_term, npos_term)
```
Comment (zhengp0) [inline src/regmod/models/tobit.py:163]: Great! @kels271828 I think here we should apply `weights` right?

Comment (kels271828) [reply src/regmod/models/tobit.py:163]: No, the weights are applied in the objective

Comment (zhengp0) [reply src/regmod/models/tobit.py:163]: Oh, sorry, the `weights` are applied in the `get_nll_terms`, I also get tripped here. Everything is good.

**Review by zhengp0 — APPROVED** [review]
Thanks @kels271828! it is good to merge.

**Pull request merged**